controlpanel/api/models/tool.py

```diff
@@ -204,3 +204,68 @@ def restart(self, id_token):
         """
 
         cluster.ToolDeployment(self.user, self.tool).restart(id_token)
+
+
+class HomeDirectoryManager:
+    """
+    Emulates a Django model manager.
+    """
+
+    def create(self, *args, **kwargs):
+        home_directory = HomeDirectory(*args, **kwargs)
+        home_directory.save()
+        return home_directory 
+
+    def filter(self, **kwargs):
+        return []
+
+
+class HomeDirectory:
+    """
+    Represents a user's home directory in the cluster
+    """
+
+    DoesNotExist = django.core.exceptions.ObjectDoesNotExist
+    Error = cluster.HomeDirectoryResetError
+    MultipleObjectsReturned = django.core.exceptions.MultipleObjectsReturned
+
+    objects = HomeDirectoryManager()
+
+    def __init__(self, user):
+        self._subprocess = None
+        self.user = user
+
+    def __repr__(self):
+        return f"<HomeDirectoryManager: {self.user!r}>"
+
+    def reset(self):
+        """
+        Update the user's home directory (asynchronous).
+        """
+        self._subprocess = cluster.User(self.user).reset()
+
+    def get_status(self):
+        """
+        Get the current status of the reset.
+        Polls the subprocess if running, else returns an "is reset" status.
+        """
+        if self._subprocess:
+            status = self._poll()
+            if status:
+                return status
+
+        return cluster.HOME_RESET
+
+    def _poll(self):
+        """
+        Poll the deployment subprocess for status
+        """
+
+        if self._subprocess.poll() is None:
+            return cluster.HOME_RESETTING
+
+        if self._subprocess.returncode:
+            log.error(self._subprocess.stderr.read().strip())
+            return cluster.HOME_RESET_FAILED
```

Comment:
these `cluster.STATUS` references may be another clue that this class may belong in the `cluster` module ☺️

Reply:
Again, just following the conventions found in the tool deploy related classes in the `tool` module. It does seem a bit odd that any of this stuff should be in this part of the code base. A sign that we should refactor perhaps..? :+1:

Reply:
I'm not sure, the whole checking for the helm release process to finish is a bit low-level. I don't have an answer to that but I haven't touched it as I'm moving few other things around and I tend to avoid changing many things at the same time.

But it was something I thought we could have a look after this Tool Catalogue bit as there may be some little bugs in the tool deployment status logic etc...

Reply:
OK... I've left as is (see incoming commit).